src/pages/Home/Header/Title/DropdownButton.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useRef } from 'react'
+import styled, { css } from 'styled-components'
+import { useToggle } from 'react-use'
+import { useSearchParams } from 'react-router-dom'
+import { useFocusOutside } from '~src/hooks/useFocusOutside'
+
+const dropdownStyles = css`
+  border-radius: 16px;
+  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.2);
+  border: none;
+  font-size: 16px;
+  padding: 10px 20px;
+  cursor: pointer;
+  transition: all 0.3s ease;
+`
+
+const StyledButton = styled.button<{ open: boolean }>`
```

Comment:
I would abstract the interface

interface IButton {
  open: boolean
}

then use it 

const StyledButton = styled.button<IButton>